### DISCUSSION TITLE
Which JS SPA client package for OIDC?

### DISCUSSION THREAD

**Discussion by idc77** [goauthentik/authentik#9664] [Q&A] [answered]
Which client package do you recommend for OIDC SPA?

Personally I use Vue with Quasar.
When I was using keycloak, it had the `keycloak-js` adapter, also `@dsb-norge/vue-keycloak-js`.

Does authentik have something like it?

**Comment by BeryJu** [comment, marked as answer]
In https://github.com/goauthentik/authentik/pull/7692 we're using `oidc-client-ts`

**Reply by idc77** [reply]
Thanks but I'm not sure I understand what this issue means.
Also I don't see oidc-client-ts in web/package.json
Ah you're using lit.
Oh well, thanks for now.

**Reply by BeryJu** [reply]
Let me rephrase the answer, we'll be using that client ourselves in the future so that's the go-to recommendation for now.

**Comment by benmarte** [comment]
You can also use Auth.js which has an integration that works great with authentik.

https://authjs.dev/getting-started/providers/authentik
